agent/js/src/wd_server.js

```diff
@@ -1028,13 +1028,34 @@ WebDriverServer.prototype.onTracingMessage_ = function(message) {
 };
 
 /**
- * Runs lighthouse with the options specified by the task and dumps data into 
+ * Runs lighthouse with the options specified by the task and dumps data into
  * a directory.
  * @private
  */
 WebDriverServer.prototype.runLighthouse_ = function() {
-  // TODO(phulce): actually run lighthouse
-  this.onPageLoad_(new Error('Lighthouse not yet implemented!'));
+  this.lighthouseFile_ = path.join(this.runTempDir_, 'lighthouse.html');
+  this.app_.timeout(1000, 'Waiting for Chrome to be available...');
+  process_utils.scheduleExec(this.app_,
+    'lighthouse',
+    [
+      '--disable-device-emulation',
+      '--disable-network-throttling',
+      '--save-assets',
+      '--port', this.browser_.devToolsPort_,
+      '--output', 'html',
+      '--output-path', this.lighthouseFile_,
+      this.task_.url,
+    ], undefined, this.timeout_).then(function () {
+    var devtoolsLogFile = path.join(this.runTempDir_, 'lighthouse-0.devtoolslog.json');

```

Comment:
this line yes :) though the above should be indented 2 more spaces. it's used elsewhere like this, will fix